spacy/kb/candidate.pxd

```diff
@@ -1,12 +1,16 @@
-from .kb cimport KnowledgeBase
 from libcpp.vector cimport vector
+from .kb_in_memory cimport InMemoryLookupKB
 from ..typedefs cimport hash_t
 
-# Object used by the Entity Linker that summarizes one entity-alias candidate combination.
 cdef class Candidate:
-    cdef readonly KnowledgeBase kb
-    cdef hash_t entity_hash
-    cdef float entity_freq
-    cdef vector[float] entity_vector
-    cdef hash_t alias_hash
-    cdef float prior_prob
+    cdef readonly str _entity_id_
```

Comment:
Ah, hm. Storing the string specifically sort of defeats the point of having the hash and storing that though...

Reply:
Ya, it's not ideal. If we don't do that though we can't support string IDs.

Reply:
Alternatively: we don't have an attribute for `entity_id` / `entity_id` in `Candidate` and leave that as an implementation detail for the inheriting class. I think that'd be fine as the `Candidate` interface would still describe all the functionality that inheriting classes have to inherit.

Reply:
I actually do think that'd be better. This would avoid the weirdness of always having to specify your `entity_id` as string even if it's an int.

Reply:
Maybe we just want to remove most of these from `Candidate` and put them in `InMemoryCandidate` itself, so that inheritance remains as flexible as possible?